python/llm/src/bigdl/llm/convert_model.py

```diff
@@ -0,0 +1,95 @@
+        invalidInputError(check, f"Invaid input kwargs found: {_used_args}")
+        ggml_convert_model(input_path=input_path,
+                           output_path=output_path,
+                           model_family=model_family,
+                           dtype=dtype,
+                           **_used_args,
+                           )
+    elif model_type == "gptq":
+        invalidInputError(input_path.endswith(".pt"), "only support pytorch's .pt format now.")
+        invalidInputError(model_family == "llama" and dtype == 'int4',
+                          "Convert GPTQ models should always "
+                          "specify `--model_family llama --dtype int4` in the command line.")
+        check, _used_args = _special_kwarg_check(kwargs=kwargs,
+                                                 check_args=["tokenizer_path"])
+        invalidInputError(check, f"Invaid input kwargs found: {_used_args}")
+        invalidInputError("tokenizer_path" in _used_args,
+                          "The GPT-Q model requires the `tokenizer_path` parameter to be provided."
+                          "Usage: convert-model --model_type gptq"
+                          "--model_family llama --input_path llamaXXb-4bit.pt"
+                          "--tokenizer_path tokenizer.model --output_path out.bin")
+        convert_gptq2ggml(input_path=input_path,
+                          tokenizer_path=_used_args["tokenizer_path"],
+                          output_path=output_path)
+    else:
+        invalidInputError(False, f"Unsupported input model_type: {model_type}")
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Model Convert Parameters')
+    parser.add_argument('-i', '--input_path', type=str, required=True,
+                        help=("input_path, a path to a *directory* containing model weights"))
+    parser.add_argument('-o', '--output_path', type=str, required=True,
+                        help=("output_path,save path of output quantized model."))
+    parser.add_argument('-x', '--model_family', type=str, required=True,
+                        help=("model_family: Which model family your input model belongs to."
+                              "Now only `llama`/`bloom`/`gptneox` are supported."))
+    parser.add_argument('-m', '--model_type', type=str, required=True,
+                        help=("The model type to be convert to a ggml compatible file."
+                              "Now only `pth`/`gptq` are supported."))
+    parser.add_argument('-t', '--dtype', type=str, default="int4",
+                        help="Which quantized precision will be converted.")
+
+    # pth specific args
+    parser.add_argument('-p', '--tmp_path', type=str, default=None,
+                        help="Which path to store the intermediate model during the"
+                        "conversion process.")
+
+    # gptq specific args
+    parser.add_argument('--tokenizer_path', type=str, default=None,
```

Comment:
it's common practice to use `-` instead of `_` in commadline arguments. llama cpp cli args also conforms to this style, e.g. `--prompt-cache` and `--prompt-cache-all`.

I think we can change `_` in args to `-`, e.g. `--model_type` -> `--model-type`, `--output_path` -> `--output-path`, etc.